include/operations/blas_constants.h

```diff
@@ -263,18 +249,4 @@ struct constant_pair {
 
 }  // namespace blas
 
-#ifndef __ADAPTIVECPP__
-template <typename ind_t, typename val_t>
-struct cl::sycl::is_device_copyable<blas::IndexValueTuple<ind_t, val_t>>
-    : std::true_type {};
-
-template <typename ind_t, typename val_t>
-struct cl::sycl::is_device_copyable<const blas::IndexValueTuple<ind_t, val_t>>
-    : std::true_type {};
-
-template <typename ind_t, typename val_t>
-struct std::is_trivially_copyable<blas::IndexValueTuple<ind_t, val_t>>
-    : std::true_type {};
-#endif
-
```

Comment:
Why are we able to get rid of this?

Reply:
Thank you for your question @hjabird.

As you see I removed also both copy constructors from `IndexValueTuple` class above. In this way the compiler generates them for us, this make the class *trivially_copyable* (as I understood  from [this definition](https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class)). This is key to remove the lines you highlighted. By SYCL2020 reference, if a class is *trivially_copyable* it is also [*device_copyable*](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec::device.copyable), which is needed for all buffers(see [this](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/buffer.hpp#L168)). These lines above force to consider `IndexValueTuple` class as *device_copyable* but removing the copy constructors needed for ComputeCpp it already has this property, so we don't need them anymore.